examples/web-chat/src/App.tsx

```diff
@@ -1,230 +1,51 @@
-  },
-};
-
-export const ChatContentTopic = "/toy-chat/2/huilong/proto";
-const ChatDecoder = new Decoder(ChatContentTopic);
-
-async function retrieveStoreMessages(
-  waku: WakuLight,
-  setArchivedMessages: (value: Message[]) => void
-): Promise<void> {
-  const startTime = new Date();
-  // Only retrieve a week of history
-  startTime.setTime(Date.now() - 1000 * 60 * 60 * 24 * 7);
-
-  const endTime = new Date();
+import { useWaku, useContentPair } from "@waku/react";
 
-  try {
-    for await (const messagesPromises of waku.store.queryGenerator(
-      [ChatDecoder],
-      {
-        pageSize: 5,
-        pageDirection: PageDirection.FORWARD,
-        timeFilter: {
-          startTime,
-          endTime,
-        },
-      }
-    )) {
-      const wakuMessages = await Promise.all(messagesPromises);
+import { useMessages, usePersistentNick } from "./hooks";
 
-      const messages: Message[] = [];
-      wakuMessages
-        .filter(isMessageDefined)
-        .map((wakuMsg) => Message.fromWakuMessage(wakuMsg))
-        .forEach((message) => {
-          if (message) {
-            messages.push(message);
-          }
-        });
-      setArchivedMessages(messages);
-    }
-  } catch (e) {
-    console.log("Failed to retrieve messages", e);
-  }
-}
+const startTime = new Date();
+// Only retrieve a week of history
+startTime.setTime(Date.now() - 1000 * 60 * 60 * 24 * 7);
+const endTime = new Date();
 
 export default function App() {
-  const [messages, dispatchMessages] = useReducer(reduceMessages, []);
-  const [waku, setWaku] = useState<WakuLight | undefined>(undefined);
-  const [nick, setNick] = useState<string>(() => {
-    const persistedNick = window.localStorage.getItem("nick");
-    return persistedNick !== null ? persistedNick : generate();
-  });
-  const [historicalMessagesRetrieved, setHistoricalMessagesRetrieved] =
-    useState(false);
-
-  useEffect(() => {
-    localStorage.setItem("nick", nick);
-  }, [nick]);
-
-  useEffect(() => {
-    initWaku(setWaku)
-      .then(() => console.log("Waku init done"))
-      .catch((e) => console.log("Waku init failed ", e));
-  }, []);
-
-  useEffect(() => {
-    if (!waku) return;
-    // Let's retrieve previous messages before listening to new messages
-    if (!historicalMessagesRetrieved) return;
-
-    const handleIncomingMessage = (wakuMsg: DecodedMessage) => {
-      console.log("Message received: ", wakuMsg);
-      const msg = Message.fromWakuMessage(wakuMsg);
-      if (msg) {
-        dispatchMessages([msg]);
-      }
-    };
-
-    let unsubscribe: undefined | (() => Promise<void>);
-    waku.filter.subscribe([ChatDecoder], handleIncomingMessage).then(
-      (_unsubscribe) => {
-        console.log("subscribed to ", ChatContentTopic);
-        unsubscribe = _unsubscribe;
+  const { node } = useWaku<LightNode>();
+  const { decoder } = useContentPair();
+  const [messages, pushLocalMessages] = useMessages({
+    node,
+    decoder,
+    options: {
+      pageSize: 5,
```

Comment:
Let's not specify the `pageSize` to make it simpler.

Reply:
will include into follow up